testng-core/src/test/java/test/retryAnalyzer/github2669/FailedRetryListener.java

```diff
@@ -0,0 +1,26 @@
+package test.retryAnalyzer.github2669;
+
+import org.testng.IAnnotationTransformer;
+import org.testng.IRetryAnalyzer;
+import org.testng.annotations.ITestAnnotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+
+public class FailedRetryListener implements IAnnotationTransformer {
+
+  @Override
+  public void transform(ITestAnnotation iTestAnnotation, Class aClass,
+                        Constructor constructor, Method method) {
+    {
+      IRetryAnalyzer retry = null;
+      try {
+        retry = iTestAnnotation.getRetryAnalyzerClass().newInstance();
+      } catch (InstantiationException | IllegalAccessException e) {
+        e.printStackTrace();
+      }
+      if (retry == null) {
+        iTestAnnotation.setRetryAnalyzer(FailedRetry.class);
```

Comment:
This can be moved within the `catch` block and the printstacktrace part can be removed